src/main/java/newbank/server/Account.java

```diff
@@ -2,24 +2,80 @@
 
 import org.javamoney.moneta.Money;
 
+import newbank.server.exceptions.AccountBalanceInvalidException;
+import newbank.server.exceptions.AccountInvalidException;
 import newbank.server.exceptions.AccountNameInvalidException;
 
+/**
+ * Representation of a customer's account.
+ *
+ * <p>An account is identified by a name and has a balance. An account also has a flag indicating
+ * whether that account is the default current account for a customer. The default current account
+ * is the account used (by default) for all transactions (for instance when receiving money from
+ * another customer or when paying money to a customer). Note that a Savings account cannot be used
+ * as the default current account.
+ */
 public class Account {
   private String accountName;
   private Money balance;
+  private boolean isDefault = false;
 
   public static final int MIN_NAME_LENGTH = 4;
   public static final int MAX_NAME_LENGTH = 12;
 
-  public Account(String accountName, Money openingBalance) throws AccountNameInvalidException {
-    this.validate(accountName, openingBalance);
+  public static class Builder {
```

Comment:
Hi Jules. Could you please justify the decision to use the Builder pattern here?

Reply:
One reason is to keep the number of arguments small. Another reason is to avoid using a boolean as parameter. Finally, it's is more readable and scale better when we add more parameters to this class.

Kindest Regards,
Jules

Reply:
> Another reason is to avoid using a boolean as parameter.
Yes, this is a very convincing use :+1:

I'm just thinking if the user of the Builder can now forget about some crucial parameters? A check inside of the final `.build` method would probably do, right?

Reply:
Correct. However in our case after refactoring the extra parameter vanished so the pattern was no longer justified, so I removed it...

Kindest Regards,
Jules